feat(nav): Update Tabbed Nav on CRUD Pages

Antonio-RiveroMartnez opened the pull request:
### SUMMARY
These changes are part of a bigger collection of changes we are introducing to navigation in our menu to direct users to more useful flows. 

- Updates to tab menus for Databases, Datasets, Saved Queries and Query History menu
- Including new tests to cover changes

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
![error](https://user-images.githubusercontent.com/38889534/186922697-539e581a-54bd-4cf4-8ce0-347f4110aa04.gif)

After:
![test](https://user-images.githubusercontent.com/38889534/186922865-91113558-4487-4524-955c-300696b4a8a0.gif)


### TESTING INSTRUCTIONS
1. Load the app
2. Check all the new changes:
```
1. Databases page no longer shows the tabbed nav for Databases / Datasets / Saved queries / Query History
2. Datasets page no longer shows the tabbed nav for Databases / Datasets / Saved queries / Query History
3. Rename page header on Databases from "Data" to "Databases"
4. Rename page header on Datasets from "Data" to "Datasets"
5. Saved query and Query history pages both show tabbed nav with the header as "SQL" in place of what used to be "Data"
6. Databases and Datasets are removed from the SQL tabbed nav (remaining menu items are Saved queries / Query history)
```

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [X] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

AAfghahi commented on superset-frontend/src/views/CRUD/data/database/DatabaseList.test.jsx:
```diff
@@ -139,6 +139,10 @@ describe('DatabaseList', () => {
   });
 
+  it('renders a SubMenu with no tabs', () => {
+    expect(wrapper.find(SubMenu).props().tabs).toBeUndefined();
```
is undefined correct here or null?

Antonio-RiveroMartnez replied:
undefined is correct since nothings should be passed to it, not even null.

AAfghahi commented on superset-frontend/src/views/CRUD/data/dataset/DatasetList.test.jsx:
```diff
@@ -181,6 +182,14 @@ describe('DatasetList', () => {
+  });
+
+  it('renders a SubMenu with no tabs', () => {
+    expect(wrapper.find(SubMenu).props().tabs).toBeUndefined();
```
same thought process here.

Antonio-RiveroMartnez replied:
undefined is correct since nothings should be passed to it, not even null.

AAfghahi reviewed (changes requested):
It looks like the CI has something wrong with it, you have the same error that Phillip did. . .Potentially just need a rebase.

Antonio-RiveroMartnez commented:
@AAfghahi I just rebased it, however, I can see all lastest PRs failing because some new changes introduced in master regarding `PivotTableChart`.

lyndsiWilliams commented:
/testenv up

Antonio-RiveroMartnez committed:
Menu:

- Updates to tab menus for Databases, Datasets, Saved Queries and Query History menu
- Including new tests to cover changes

Antonio-RiveroMartnez commented:
@AAfghahi all rebased and passing the checks now.

lyndsiWilliams commented:
/testenv up

lyndsiWilliams reviewed (approved):
lgtm!

yousoph commented:
/testenv up

yousoph commented:
/testenv up

eschutho reviewed (approved):
nice tests, too!
